app/lib/actions/katello/activation_key/update.rb

```diff
@@ -15,15 +15,27 @@ module Katello
     module ActivationKey
       class Update < Actions::EntryAction
         def plan(activation_key, activation_key_params)
+          cp_call = cp_changed?(activation_key.auto_attach, activation_key_params[:auto_attach]) ||
+                    cp_changed?(activation_key.service_level, activation_key_params[:service_level]) ||
+                    cp_changed?(activation_key.release_version, activation_key_params[:release_version])
+
           activation_key.disable_auto_reindex!
           activation_key.update_attributes!(activation_key_params)
-          plan_action(::Actions::Candlepin::ActivationKey::Update,
-                      cp_id: "activation_key.cp_id",
-                      release_version: "activation_key.releaseVer",
-                      service_level: "activation_key.serviceLevel",
-                      auto_attach: "activation_key.autoAttach")
+
+          if cp_call
+            plan_action(::Actions::Candlepin::ActivationKey::Update,
+                        cp_id: activation_key.cp_id,
+                        release_version: activation_key.release_version,
+                        service_level: activation_key.service_level,
+                        auto_attach: activation_key.auto_attach)
+          end
+
           action_subject activation_key
         end
+
+        def cp_changed?(activation_key, activation_key_params)
+          return !activation_key_params.nil? && activation_key.to_s != activation_key_params.to_s
```

Comment:
the `return` is not needed here, ruby will just automatically return the result of this statement